spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -813,17 +805,14 @@ def _finalise_restart(self, reset=False):
             sw.kernel_manager.autorestart = True
             sw.kernel_manager.start_restarter()
 
-            # For spyder-ide/spyder#6235, IPython was changing the
-            # setting of %colors on windows by assuming it was using a
-            # dark background. This corrects it based on the scheme.
-            self.set_color_scheme(sw.syntax_style, reset=reset)
+            if reset:
+                sw.reset(clear=True)
             sw._append_html(_("<br>Restarting kernel...<br>"),
                             before_prompt=True)
             sw.insert_horizontal_ruler()
 
-            self.send_kernel_configuration()
+            self.send_spyder_kernel_configuration()
 
-        self._hide_loading_page()
```

Comment:
Why is this no longer necessary?

Reply:
I think it was never necessary, because the loading page will be hidden as soon as a prompt comes. This just closes the loading page prematurely as far as I can tell